From: Mitko Haralanov <mitko.haralanov@intel.com>
Subject: IB/hfi1: Prevent NULL pointer dereference
Git-commit: 4b00d9490f2147d29f107f36391b0cc77bcd944f
Patch-mainline: v4.6
References: bsc#973818, fate#319242
Acked-by: Jiri Bohac <jbohac@suse.cz>

Prevent a potential NULL pointer dereference (found
by code inspection) when unregistering an MMU handler.

Reviewed-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Reviewed-by: Dean Luick <dean.luick@intel.com>
Signed-off-by: Mitko Haralanov <mitko.haralanov@intel.com>
Signed-off-by: Jubin John <jubin.john@intel.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 drivers/staging/rdma/hfi1/mmu_rb.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/staging/rdma/hfi1/mmu_rb.c
+++ b/drivers/staging/rdma/hfi1/mmu_rb.c
@@ -107,6 +107,9 @@ void hfi1_mmu_rb_unregister(struct rb_ro
 	struct mmu_rb_handler *handler = find_mmu_handler(root);
 	unsigned long flags;
 
+	if (!handler)
+		return;
+
 	spin_lock_irqsave(&mmu_rb_lock, flags);
 	list_del(&handler->list);
 	spin_unlock_irqrestore(&mmu_rb_lock, flags);
